CactbotOverlay/FFXIVProcess.cs

```diff
@@ -570,525 +504,445 @@ public class EntityData {
+                return JObject.FromObject(*(SummonerJobMemory*)&p[0]);
+              };
+            case EntityJob.SCH: {
+                return JObject.FromObject(*(ScholarJobMemory*)&p[0]);
+              };
+            case EntityJob.MNK: {
+                return JObject.FromObject(*(MonkJobMemory*)&p[0]);
+              };
+            case EntityJob.MCH: {
+                return JObject.FromObject(*(MachinistJobMemory*)&p[0]);
+              };
+            case EntityJob.AST: {
+                return JObject.FromObject(*(AstrologianJobMemory*)&p[0]);
+              };
+            case EntityJob.SAM: {
+                return JObject.FromObject(*(SamuraiJobMemory*)&p[0]);
+              };
+            default: {
+                logger_.LogError("GetJobSpecificData: {0}", job);
+                return null;
+              };
+          }
+        }
       }
     }
 
-    public RedMageJobData GetRedMage() {
-      byte[] bytes = GetJobSpecificData();
-      if (bytes == null)
-        return null;
+    [Serializable]
+    [StructLayout(LayoutKind.Explicit)]
+    public struct RedMageJobMemory {
+      [FieldOffset(0x00)]
+      public byte whiteMana;
 
-      var j = new RedMageJobData();
-      j.white = bytes[kJobDataInnerStructOffsetJobSpecificData];
-      j.black = bytes[kJobDataInnerStructOffsetJobSpecificData + 1];
-      return j;
-    }
+      [FieldOffset(0x01)]
+      public byte blackMana;
+    };
 
-    public class WarriorJobData {
-      public int beast = 0;
+    [Serializable]
+    [StructLayout(LayoutKind.Explicit)]
+    public struct WarriorJobMemory {
+      [FieldOffset(0x00)]
+      public byte beast;
+    };
 
-      public override bool Equals(object obj) {
-        var o = obj as WarriorJobData;
-        return o != null &&
-          beast == o.beast;
-      }
+    [Serializable]
+    [StructLayout(LayoutKind.Explicit)]
+    public struct DarkKnightJobMemory {
+      [FieldOffset(0x00)]
+      public byte blood;
+      [FieldOffset(0x02)]
+      public ushort darksideMilliseconds;
+    };
 
-      public override int GetHashCode() {
-        int hash = 17;
-        hash = hash * 31 + beast.GetHashCode();
-        return hash;
-      }
-    }
+    [Serializable]
+    [StructLayout(LayoutKind.Explicit)]
+    public struct PaladinJobMemory {
+      [FieldOffset(0x00)]
+      public byte oath;
+    };
 
-    public WarriorJobData GetWarrior() {
-      byte[] bytes = GetJobSpecificData();
-      if (bytes == null)
-        return null;
+    [Serializable]
+    [StructLayout(LayoutKind.Explicit)]
+    public struct GunbreakerJobMemory { // TODO: Needs a lot more research. 
+      [FieldOffset(0x00)]
+      public byte cartridges;
 
-      var j = new WarriorJobData();
-      j.beast = bytes[kJobDataInnerStructOffsetJobSpecificData];
-      return j;
-    }
+      [FieldOffset(0x02)]
+      public ushort continuationMilliseconds; // Notes: Jumps to 15000 (ms?) after using Gnashing Fang, doesn't decrease.
```

Comment:
continuationMilliseconds is 15000 if and only if continuationState is not zero.  I'm not sure it's milliseconds, and probably isn't worth including.

continuationState is correct.  It's 0 when not doing any combos, it's 1 after you hit Gnashing Fang, and 2 after you hit Savage Claw.  When you hit Wicked Talon it goes back to 0.